src/api/ruleengine.ts

```diff
@@ -125,7 +125,11 @@ export function deleteRules(id: string): Promise<any> {
 
 export function testsql(body: Record<string, unknown>): Promise<any> {
   return http.post('/rule_test', body, {
-    transformResponse: [(data: string) => data],
+    transformResponse: [
+      (data: string, responseHeader: any, code: number) => {
+        return code > 299 ? JSON.parse(data) : data
```

Comment:
Give the 299 a clear name, or leave some comments here to inform others about the number.